DANS CE PROJET NOUS ALLONS ETUDIER LES DONNEES DE LA BANQUE MONDIALE. 

EN PARTICULIER LES DONNEES STATISTIQUES SUR L'EDUCATION. 


In [4091]:
# COMMENCONS PAR IMPORTER LES LIBRAIRIES NECESSAIRES 
import numpy as np # POUR LE CALCUL MATRICIEL NOTAMMENT
import pandas as pd # POUR LES DATAFRAMES
import matplotlib.pyplot as plt # POUR LES REPRESENTATIONS GRAPHIQUES

import seaborn as sns # POUR CREER DES HEATMAP ET AUTRE

DANS LE DOSSIER A TELECHARGER SE TROUVENT 5 FICHIERS .CSV. NOUS ALLONS LES ETUDIER UN PAR UN

# ***<u> P1 - 1) ETUDE DU PREMIER FICHIER : EdStatsCountry</u>***

In [4092]:
# DEFINITION DU PATH
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsCountry.csv'

In [4093]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Country = pd.read_csv(path)

In [4094]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Country.head(2)

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   

            IMF data dissemination standard  Latest population census  \
0                                       NaN                      2010   
1  General Data Dissemination System (GDDS)                      1979   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   

  Source of most recent Income and expenditure data  \
0                                               NaN   
1           Integrated household survey (IHS), 2008   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   

  Unnamed: 31  
0         NaN  
1         NaN  

[2 rows x 32 columns]

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE

IL COMPORTE 32 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

LES DIFFERENTS NOMS (SHORT, TABLE, LONG)

LE PIB (HIGH, LOW ...)

DES DATES DE RECENSEMENT DE LA POPULATION ET DES PARCELLES AGRICOLES ET AUTRES


In [4095]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Country.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

CE FICHIER CONTIENT  241 LIGNES NUMEROTEES DE 0 A 240 ET 32 COLONNES NUMEROTEES DE 0 A 31. IL COMPORTE 4 COLONNES DE TYPES FLOAT (NOMBRES DECIMAUX) ET 28 DE TYPE OBJECT (CHAINE DE CARACTERES). 

ON CONSTATE QUE LA COLONNE N° 31 (Unnamed: 31) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. D'AUTRES COMME LA N°16 (Alternative conversion factor) OU ENCORE LA N°11 (National accounts reference year).... NE SONT QUE PEU REMPLIES. D'AUTRES COMME LA N°5 (Currency Unit) SONT QUASIMENT COMPLETES. ENFIN DES COLONNES TELLES QUE LA N°1 (Short Name) sont complétes. 

In [4096]:
Fichier_Country['Vital registration complete'].unique()

array(['Yes', nan, 'Yes. Vital registration for Guernsey and Jersey.'],
      dtype=object)

In [4097]:
# DANS UN PREMIER TEMPS ON NE VA CONSERVER QUE LES PAYS QUI ONT UN RENSEIGNEMENT DIF DE NAN DANS LA COLONNE VITAL REGISTRATION COMPLETE
# CETTE COLONNE INDIQUANT LA QUALITE DES DEMARCHES EFFECTUEES A L'ETAT CIVIL (NAISSANCE,DECES, MARIAGE ...)
# CE CHOIX ME SEMBLE PERTINENT AFIN DE DISPOSER ENSUITE DE VALEURS CORRECTES
Fichier_Country=Fichier_Country.loc[(Fichier_Country['Vital registration complete']=='Yes. Vital registration for Guernsey and Jersey.') | (Fichier_Country['Vital registration complete']=='Yes')]

In [4098]:
# ON VA DROP TOUTES LES COLONNES QUI ONT PLUS DE 30% DE NAN

Liste_à_drop = []
for features in Fichier_Country.columns:
    if Fichier_Country[features].isnull().sum()>0.3*Fichier_Country.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Country=Fichier_Country.drop(Liste_à_drop,axis=1)
Fichier_Country.info()

['Special Notes', 'National accounts reference year', 'Lending category', 'Other groups', 'Alternative conversion factor', 'PPP survey year', 'External debt Reporting status', 'Latest household survey', 'Source of most recent Income and expenditure data', 'Latest agricultural census', 'Latest industrial data', 'Unnamed: 31']
<class 'pandas.core.frame.DataFrame'>
Index: 111 entries, 0 to 232
Data columns (total 20 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Country Code                       111 non-null    object 
 1   Short Name                         111 non-null    object 
 2   Table Name                         111 non-null    object 
 3   Long Name                          111 non-null    object 
 4   2-alpha code                       110 non-null    object 
 5   Currency Unit                      111 non-null    object 
 6   Region                             111 non-null    objec

# ***<u> P1 - 2) PASSONS AU PROCHAIN FICHIER NOMME EdStatsCountry-Series</u>***


In [4099]:
# DEFINITION DU PATH
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsCountry-Series.csv'

In [4100]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Country_Series = pd.read_csv(path)

In [4101]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Country_Series.head(5)

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE

IL COMPORTE 4 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

UNE COLONNE NOMMEE SeriesCode EN LIEN CERTAINEMENT AVEFC LE FICHIER EdStatsSeries

UNE COLONNE DESCRIPTION INDIQUANT LA SOURCE DES INFOS. ON CONSTATE QUE LES SOURCES VARIENT. CERTAINES OFFICIELLES COMME L'ONU. D'AUTRES OBTENUES PAR REGRESSION

UNE COLONNE Unnamed: 3 CERTAINEMENT ENTIEREMENT VIDE COMME SUR LE FICHIER PRECEDENT

In [4102]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Country_Series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


CE FICHIER CONTIENT  ***613 LIGNES*** NUMEROTEES DE 0 A 612 ET ***4 COLONNES*** NUMEROTEES DE 0 A 3. IL COMPORTE *****1 COLONNE DE TYPES FLOAT***** (NOMBRES DECIMAUX) ET ***3 DE TYPE OBJECT*** (CHAINE DE CARACTERES).

ON CONSTATE QUE LA COLONNE N° 3 (Unnamed: 3) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. QUANT AU RESTE DES COLONNES ELLES SONT COMPLETES. 

In [4103]:
# ON VA DROP Unnamed: 3

Fichier_Country_Series=Fichier_Country_Series.drop(['Unnamed: 3'],axis=1)



# ***<u> P1 - 3) PASSONS AU PROCHAIN FICHIER NOMME EdStatsData</u>***



In [4104]:
# DEFINITION DU PATH
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsData.csv'

In [4105]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Data = pd.read_csv(path)

In [ ]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Data.head()

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY NAME, CEPENDANT ICI IL SEMBLE PLUTOT S'AGIR DE GROUPE GEOGRAPHIQUE COMME Arab World (Monde Arabe) OU ENCORE East Asia & Pacific (Espace Asie Pacifique). IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

IL COMPORTE 70 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

DEUX COLONNES NOMMEES Country Name ET Country Code QUI REPRESENTENT EN FAIT LES CODE DE REGIONS GEOGRAPHIOQUES AINSI QUE LEUR VERSIONS ABREGEES (EX : Arab World -> ARB)

DEUX COLONNES NOMMEES Indicator Name ET Indicator Code QUI SONT L'INDICATEUR ETUDIE AINSI QUE SON ABREVIATION (EX : Adujstment... -> UIS.NERA.2)

LA DERNIERE NOMMEE Unnamed: 69 CERTAINEMENT VIDE COMME SUR LES FICHIERS PRECEDENT

QUANT AUX 65 COLONNES RESTANTES ELLES REPRESENTENT TOUTES UNE ANNE DIFFERENTES CONTENANT LES VALEURS CONNUES POUR L'INDICATEUR PRECISE DANS LA COLONNE Indicator Name A LA DATE INDIQUEE PAR LA COLONNE. LES ANNEES AVANCENT D'UN PAS DE 1 DE 1970 A 2017 INCLUS PUIS AVANCENT D'UN PAS DE 5 DE 2020 A 2100 INCLUS.




In [ ]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Data.info()


CE FICHIER CONTIENT  886930 LIGNES NUMEROTEES DE 0 A 886929 ET 70 COLONNES NUMEROTEES DE 0 A 69. IL COMPORTE 66 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 4 DE TYPE OBJECT (CHAINE DE CARACTERES). 



ON CONSTATE QUE LA COLONNE N° 69 (Unnamed: 69) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. LA COLONNE N°51 REPRESENTANT L'ANNEE 2017 EST TRES PEU RENSEIGNEE (SEULEMENT 143 NON-NULL).LES ANNEES TERMINANT PAR 0 OU 5 ENTRE 1970 ET 1990 INCLUS SONT MIEUX RENSEIGNEES QUE LES AUTRES SUR LA MEME PERIODE. CE PHENOMENE CE REPETE DE 1990 A 2000 BIEN QUE SUR CETTE PERIODE LES "FAIBLES" RENSEIGNEMENTS ONT OTUT DE MEME DOUBLES EN VALEUR ABSOLUE. A PARTIR DES ANNEES 1999-2000 JUSQU'EN 2015 LES RENSEIGNEMENTS VONT DE 120K A 200K VOIR 240K. 2016 ET 2017 ETANT TRES MAL RENSEIGNEES LES COLONNES N°52 JUSQU'A 68 c.à.d. DE L'ANNEE 2020 JUSQU'A L'ANNEE 2100 COMPORTE TOUTES 51436 VALEURS NON NULLES (CERTAINEMENT DES PROJECTIONS DE VALEURS)


In [ ]:
list(Fichier_Data['Indicator Name'].unique())

In [ ]:
list(Fichier_Data['Indicator Code'].unique())

In [ ]:
Fichier_Data['Country Name'].unique()

ON CONSTATE QUE DANS CETTE TABLE LE COUNTRY CODE DESIGNE DES PAYS MAIS AUSSI DES ZONES REGIONALES (COMME ASIE DE L'EST ET PACIFIQUE (AVEC LES LOW INCOME INCLUS OU NON), PAYS DE L'OCDE, ZONE EURO...) VOIR LA TOTALITE DU MONDE.


In [ ]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 5% DE NON NULL

Liste_à_drop = []
for features in Fichier_Data.columns:
    if Fichier_Data[features].isnull().sum()>0.95*Fichier_Data.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Data=Fichier_Data.drop(Liste_à_drop,axis=1)
Fichier_Data.info()

In [ ]:
# ON VA MAINTENANT DROP LES LIGNES TROP PEU RENSEIGNEES 
# J'AI CHOISI UNE LIMITE A AU MOINS 22 NON NULL DANS LA LIGNE AFIN D'INCLURE LES LIGNES QUI ONT AU MOINS UNE DONNEE SUPPLEMENTAIRE
# QUE LES 4 PREMIERES(QUI NE MANQUENT NULLE PART) ET LES 17 DERNIERES QUI SONT DES PROJECTIONS
Fichier_Data=Fichier_Data.dropna(thresh=22)
Fichier_Data.info()

In [ ]:
Fichier_Data.head()

# ***<u> P1 - 4) PASSONS AU PROCHAIN FICHIER NOMME EdStatsFootNote</u>***

In [4003]:
# DEFINITION DU PATH
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsFootNote.csv'

In [4004]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_FootNote = pd.read_csv(path)

In [ ]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_FootNote.head(5)


DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE, ABW POUR ARUBA COMME ON A PU LE VOIR DANS LE 1ER FICHIER. IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

IL COMPORTE 5 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

UNE COLONNE CountryCode REFERENCANT LA ZONE GEOGRAPHIQUE

UNE COLONNE NOMMEE SeriesCode REFERENCANT LE CODE DE LA SERIE DE DONNEES

UNE COLONNE Year RENSEIGNANT L'ANNEE A LAQUELLE SE RAPPORTE LA SERIE

UNE COLONNE DESCRIPTION EXPLIQUANT SI LA DONNEE EST UNE ESTIMATION OU UNE DONNEE COLLECTEE

LA DERNIERE NOMMEE Unnamed: 4 CERTAINEMENT VIDE COMME SUR LES FICHIERS PRECEDENT


In [ ]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_FootNote.info()



CE FICHIER CONTIENT  643638 LIGNES NUMEROTEES DE 0 A 643637 ET 5 COLONNES NUMEROTEES DE 0 A 4. IL COMPORTE 1 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 4 DE TYPE OBJECT (CHAINE DE CARACTERES). 

ON CONSTATE QUE LA COLONNE N° 4 (Unnamed: 4) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. QUANT AUX AUTRES COLONNE ELLES SONT TOUTES COMPLETES


In [4007]:
# ON VA DROP la colonne Unnamed: 4

Fichier_FootNote=Fichier_FootNote.drop(['Unnamed: 4'],axis=1)

# ***<u> P1 - 5) PASSONS AU PROCHAIN FICHIER NOMME EdStatsSeries</u>***

In [4008]:
# DEFINITION DU PATH
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsSeries.csv'

In [4009]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Series = pd.read_csv(path)

In [ ]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Series.head(2)


DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE Series Code, IL SEMBLERAIT QUE DANS LE NOM SE TROUVE LA TRNACHE D'AGE ETUDIE PAR EXEMPLE 1519 POUR LES 15-19 ANS. IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

L'ENSEMBLE DES COLONNES A POUR BUT DE RENSEIGNER LA MANIERE DONT ONT ETE COLLECTEES LES DONNEES, LES SOURCES UTILISEES, LES UNITES DE MESURES...

In [ ]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Series.info()


CE FICHIER CONTIENT  3665 LIGNES NUMEROTEES DE 0 A 3664 ET 21 COLONNES NUMEROTEES DE 0 A 20. IL COMPORTE 6 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 15 DE TYPE OBJECT (CHAINE DE CARACTERES). 
A NOTER QUE TOUTES LES COLONNES DE TYPES FLOAT SONT ENTIEREMENT VIDES


LES COLONNES N° 0, 1, 2, 4, 13 SONT COMPLETES. LA COLONNE N°3 Short Definition EST QUANT A ELLE COMPLETES A ENVIRON 60%. QUANT AUX AUTRES ELLES NE DEPASSENT PAS LES 552 RENSEIGNEMENTS (ENVIRON 15%). ENFIN LES N° 5, 11, 17, 18, 19, 20 SONT ENTIEREMENT VIDES

In [ ]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_Series.columns:
    if Fichier_Series[features].isnull().sum()>0.3*Fichier_Series.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Series=Fichier_Series.drop(Liste_à_drop,axis=1)
Fichier_Series.info()

# ***<u> P1 - 6) FIN ETUDE PRELIMINAIRE</u>***

# ***<u> P2 - 1) SUITE DE L'ETUDE</u>***

REPRISE DE L'ANALYSE

LE FICHIER "Fichier_Series" NE SEMBLE PAS PERTINENT POUR L'ANALYSE (IL DETAILLE EN FAIT LA METHODOLOGIE APPLIQUEE POUR RECUEUILLIR L'ENSEMBLE DES DONNEES)

DE MEME POUR "Fichier_Footnote" AINSI QUE "Fichier_Country_Series" QUI LA AUSSI SONT DES FICHIERS DE METHODOLOGIE.

NOUS ALLONS REGROUPER LES INFORMATIONS FOURNIES DANS LE DATAFRAME "Fichier_Data" DES ANNEES 2011 A 2015 AFIN D'AVOIR DES DONNEES EXPLOITABLES

In [ ]:
# AGREGATION DES VALEURS DE 2011 A 2015
Fichier_Data['Periode 2011-2015']=Fichier_Data[['2011', '2012', '2013', '2014', '2015']].sum(axis=1)
Fichier_Data.tail()

In [ ]:
# ON VA MAINTENANT CREER UN DATAFRAME DONT LES COLONNES SERONT "COUNTRY CODE"
# AINSI QUE CHAQUE INDICATOR Name PRESENTS DANS LA COLONNE "INDICATOR NAME" ACTUELLE.
# POUR LES VALEURS DE CES COLONNES NOUS UTILISERONS LES VALEURS "PERIODE 2011-2015"

DF_pivot=Fichier_Data.pivot(index='Country Code',columns='Indicator Name',values='Periode 2011-2015')
DF_pivot=DF_pivot.reset_index()

DF_pivot


ON A MAINTENANT UN DF CONTENANT 241 LIGNES (UNE PAR COUNTRY CODE) ET 1049 COLONNES (DIVERS INDICATEURS)

VERIFIONS QUE LES DONNEES SONT AU BON ENDROIT

In [ ]:
Fichier_Data.loc[(Fichier_Data['Indicator Name']=='Adjusted net intake rate to Grade 1 of primary education, both sexes (%)') 
                  & (Fichier_Data['Country Code']=='ZMB'),'Periode 2011-2015']

In [ ]:
DF_pivot.loc[DF_pivot['Country Code']=='ZMB','Adjusted net intake rate to Grade 1 of primary education, both sexes (%)']

ON RETROUVE LES MEMES VALEURS PARFAIT

In [4017]:

# CALCULER LE POURCENTAGE DE NAN POUR CHAQUE COLONNES
POURCENTAGE_NAN = DF_pivot.isnull().mean() * 100


ICI DF_PIVOT.ISNULL() EST UN DATA FRAME DE MEME TAILLE QUE DF_PIVOT COMPOSE DE BOOLEENS TRUE SI LA VALEUR EST NAN FALSE SINON

ENSUITE .MEAN() CALCULE LA MOYENNE PAR COLONNE (1 POUR TRUE ET 0 POUR FALSE)

EN MULTIPLIANT CETTE MOYENNE PAR 100 ON OBTIENT LE POURCENTAGE DE NAN PRESENTS DANS CHAQUE COLONNE

In [4018]:

# DEFINITION DE LA LIMITE DE NAN ACCEPTABLE
limite = 30


In [ ]:
# VISUALISATION AVEC UN HISTOGRAMME
plt.figure(figsize=(12, 6))
plt.hist(POURCENTAGE_NAN, bins=10, edgecolor='black',color='blue')

plt.axvline(limite, color='r', linestyle='--', linewidth=2, label=f'Limite ({limite}%)')
plt.title("Distribution des pourcentages de NaN ")
plt.xlabel("Pourcentage de NaN",fontsize=14)


plt.legend()

ax=plt.gca() # commande permettant de créer un objet ax pour modifier les axes

ax.spines['top'].set_visible(False)
ax.spines['left'].set_visible(False)
ax.spines['right'].set_visible(False)

patches=plt.hist(POURCENTAGE_NAN, bins=10, edgecolor='black',color='blue')[2] # collection des barres retournée par plt.hist

ax.bar_label(patches, label_type='edge', fontsize=12)

ax.yaxis.set_visible(False)
plt.show()



In [ ]:
# NOUS ALLONS MAINTENANT DROP LES COLONNES CONTENANT PLUS DE 30% DE NAN

Liste_à_drop=[]
for features in DF_pivot.columns:
    if DF_pivot[features].isnull().sum()>0.3*DF_pivot.shape[0] and features not in ['Enrolment in secondary education, both sexes (number)',
 'Enrolment in secondary vocational, both sexes (number)',
 'Enrolment in tertiary education, all programmes, both sexes (number)',
 'Net enrolment rate, secondary, gender parity index (GPI)']:
        Liste_à_drop.append(features)
Nbre_index_drop=len(Liste_à_drop)
print(f"On va retirer {Nbre_index_drop} indicateurs")
DF_pivot=DF_pivot.drop(Liste_à_drop,axis=1)
DF_pivot.info()

In [ ]:
DF_pivot

ON CONSTATE QUE CERTAINS PAYS ONT ENCORE UN GRAND NOMBRE DE RENSEIGNEMENTS MANQUANTS

VERIFIONS CA


In [ ]:

# Calculer le pourcentage de NaN pour chaque pays
POURCENTAGE_NAN_LIGNE = DF_pivot.groupby('Country Code').apply(lambda x: x.isna().mean() * 100).mean(axis=1).sort_values(ascending=False)
POURCENTAGE_NAN_LIGNE

In [ ]:

# Créer la figure et les axes
fig, ax = plt.subplots(figsize=(12, 6))

# Créer le diagramme en barres
sns.barplot(x=POURCENTAGE_NAN_LIGNE.index, y=POURCENTAGE_NAN_LIGNE.values, ax=ax)



# Personnaliser le graphique
ax.set_title('Pourcentage de NaN par Pays', fontsize=16)
ax.set_xlabel('Code Pays', fontsize=12)
ax.set_ylabel('Pourcentage de NaN', fontsize=12)
ax.tick_params(axis='x', rotation=90)  # Rotation des étiquettes de l'axe x

# Ajouter une légende
ax.legend()

# Ajuster la mise en page
plt.tight_layout()

# Afficher le graphique
plt.show()

In [ ]:
POURCENTAGE_NAN_LIGNE = DF_pivot.groupby('Country Code').apply(lambda x: x.isna().mean() * 100).mean(axis=1).sort_values()
# Créer la figure et les axes
fig, ax = plt.subplots(figsize=(12, 6))

# Créer le diagramme en barres
sns.barplot(x=POURCENTAGE_NAN_LIGNE.index, y=POURCENTAGE_NAN_LIGNE.values, ax=ax)



# Personnaliser le graphique
ax.set_title('Pourcentage de NaN par Pays', fontsize=16)
ax.set_xlabel('Code Pays', fontsize=12)
ax.set_ylabel('Pourcentage de NaN', fontsize=12)
ax.tick_params(axis='x', rotation=90)  # Rotation des étiquettes de l'axe x

# Ajouter une légende
ax.legend()

# Ajuster la mise en page
plt.tight_layout()

# Afficher le graphique
plt.show()

ON PEUT VOIR SUR CE GRAPHIQUE QU'UNE PARTIE DES PAYS SEMBLENT N'AVOIR AUCUNE INFO MANQUANTES. NOUS ALLONS NOUS CONCENTRER SUR CES DERNIERS

In [ ]:
# Calculer le pourcentage de NaN pour chaque pays
POURCENTAGE_NAN_LIGNEs = DF_pivot.groupby('Country Code').apply(lambda x: x.isna().mean() * 100).mean(axis=1)


# Compter le nombre de pays complets
pays_complet = np.sum(POURCENTAGE_NAN_LIGNE <= 1)


print(f"Nombre de pays complet: {pays_complet}")


IL Y A DONC 122 PAYS COMPLETS. NOUS ALLONS LES CONSERVER ET ELIMINER LES AUTRES

In [4026]:
# DROP DES LIGNES AVEC AU MOINS UN NAN
DF_pivot=DF_pivot.dropna(axis=0,how='any')



In [ ]:
DF_pivot

In [ ]:
DF_pivot=DF_pivot.reset_index()
DF_pivot

In [ ]:
DF_pivot=DF_pivot.drop(['index'], axis=1)
DF_pivot

In [4030]:
DF_pivot=DF_pivot.drop(['Enrolment in primary education, both sexes (number)',
       'Enrolment in primary education, female (number)','Enrolment in secondary general, female (number)','Theoretical duration of pre-primary education (years)',
       'Theoretical duration of primary education (years)','Unemployment, female (% of female labor force)',
       'Unemployment, male (% of male labor force)','Population, male', 'Population, male (% of total)','School life expectancy, primary, female (years)',
       'School life expectancy, primary, male (years)','Population, ages 7-9, male', 'Population, ages 7-9, total',
       'Population, female', 'Population, female (% of total)','Population, ages 7-13, female', 'Population, ages 7-13, male','Population, ages 7-9, female',
       'Population, ages 7-12, female', 'Population, ages 7-12, male','Population, ages 7-11, male','Population, ages 7-11, female',
       'GDP at market prices (constant 2005 US$)',
       'GDP at market prices (current US$)',
       'GDP per capita (constant 2005 US$)', 'GDP per capita (current US$)',
       'GDP per capita, PPP (constant 2011 international $)',
       'GDP per capita, PPP (current international $)',
       'GDP, PPP (constant 2011 international $)','GNI (current US$)',
       'GNI per capita, Atlas method (current US$)',
       'GNI per capita, PPP (current international $)',
       'GNI, PPP (current international $)','Labor force, female (% of total labor force)', 'Labor force, total','Mortality rate, under-5 (per 1,000)',
       'Official entrance age to lower secondary education (years)',
       'Official entrance age to pre-primary education (years)','Official entrance age to primary education (years)',
       'Percentage of students in primary education who are female (%)','Percentage of students in secondary general education who are female (%)',
       'Population growth (annual %)','Population of the official age for lower secondary education, both sexes (number)',
       'Population of the official age for lower secondary education, female (number)',
       'Population of the official age for lower secondary education, male (number)',
       'Population of the official age for pre-primary education, both sexes (number)','Population of the official age for pre-primary education, female (number)',
       'Population of the official age for pre-primary education, male (number)','Population of the official age for primary education, both sexes (number)',
       'Population of the official age for primary education, female (number)','Population, ages 7-11, total',
       'Population, ages 7-12, total', 'Population, ages 7-13, total','Population, ages 6-9, total', 'Population, ages 7-10, female',
       'Population, ages 7-10, male','Population of the official age for primary education, male (number)',
       'Population of the official age for secondary education, female (number)',
       'Population of the official age for secondary education, male (number)','Population, ages 6-12, total', 'Population, ages 6-9, female',
       'Population, ages 6-9, male','Population of the official age for tertiary education, female (number)','Population, ages 6-11, total', 'Population, ages 6-12, female',
       'Population, ages 6-12, male','Population, ages 6-10, total', 'Population, ages 6-11, female',
       'Population, ages 6-11, male','Gross enrolment ratio, primary, both sexes (%)',
       'Gross enrolment ratio, primary, female (%)',
       'Gross enrolment ratio, primary, gender parity index (GPI)',
       'Gross enrolment ratio, primary, male (%)','Population, ages 5-9, total', 'Population, ages 6-10, female',
       'Population, ages 6-10, male','Population of the official age for tertiary education, male (number)',
       'Population of the official age for the last grade of primary education, both sexes (number)',
       'Population of the official age for the last grade of primary education, female (number)',
       'Population of the official age for the last grade of primary education, male (number)','Population, ages 5-11, total', 'Population, ages 5-9, female',
       'Population, ages 5-9, male','Population of the official age for upper secondary education, female (number)',
       'Population of the official age for upper secondary education, male (number)',
       'Population of the official entrance age to primary education, both sexes (number)','Population, ages 5-10, total', 'Population, ages 5-11, female',
       'Population, ages 5-11, male','Population of the official entrance age to primary education, female (number)',
       'Population of the official entrance age to primary education, male (number)','Population, ages 4-6, total', 'Population, ages 5-10, female',
       'Population, ages 5-10, male','Population of the official entrance age to secondary general education, female (number)',
       'Population of the official entrance age to secondary general education, male (number)','Population, ages 3-5, total', 'Population, ages 4-6, female',
       'Population, ages 4-6, male','Population, age 0, female', 'Population, age 0, male','Population, ages 15-64, total', 'Population, ages 3-5, female',
       'Population, ages 3-5, male','Population, age 0, total', 'Population, age 1, female','Population, ages 15-64 (% of total)', 'Population, ages 15-64, female',
       'Population, ages 15-64, male','Population, age 1, male', 'Population, age 1, total','Population, ages 15-24, female', 'Population, ages 15-24, male',
       'Population, ages 15-24, total','Population, age 10, female', 'Population, age 10, male','Population, ages 14-19, female', 'Population, ages 14-19, male',
       'Population, ages 14-19, total','Population, age 10, total', 'Population, age 11, female','Population, ages 14-18, female', 'Population, ages 14-18, male',
       'Population, ages 14-18, total','Population, age 11, male', 'Population, age 11, total','Population, ages 13-19, female', 'Population, ages 13-19, male',
       'Population, ages 13-19, total','Population, age 12, female', 'Population, age 12, male','Population, ages 13-18, female', 'Population, ages 13-18, male',
       'Population, ages 13-18, total','Population, age 12, total', 'Population, age 13, female','Population, ages 13-17, female', 'Population, ages 13-17, male',
       'Population, ages 13-17, total','Theoretical duration of lower secondary education (years)','Population, age 13, male', 'Population, age 13, total',
       'Population, ages 12-18, total', 'Population, ages 13-16, female',
       'Population, ages 13-16, male', 'Population, ages 13-16, total','Population, age 14, female', 'Population, age 14, male',
       'Population, ages 12-17, male', 'Population, ages 12-17, total',
       'Population, ages 12-18, female', 'Population, ages 12-18, male','Population, age 14, total', 'Population, age 15, female',
       'Population, ages 12-16, female', 'Population, ages 12-16, male',
       'Population, ages 12-16, total', 'Population, ages 12-17, female',
       'Population, age 15, male', 'Population, age 15, total','Population, ages 11-18, total', 'Population, ages 12-15, female',
       'Population, ages 12-15, male', 'Population, ages 12-15, total','Population, age 16, female', 'Population, age 16, male',
       'Population, age 16, total', 'Population, age 17, female',
       'Population, age 17, male', 'Population, age 17, total',
       'Population, age 18, female', 'Population, age 18, male',
       'Population, age 18, total', 'Population, age 19, female',
       'Population, age 19, male', 'Population, age 19, total',
       'Population, age 2, female', 'Population, age 2, male',
       'Population, age 2, total', 'Population, age 20, female',
       'Population, age 20, male', 'Population, age 20, total',
       'Population, age 21, female', 'Population, age 21, male',
       'Population, age 21, total', 'Population, age 22, female',
       'Population, age 22, male', 'Population, age 22, total',
       'Population, age 23, female', 'Population, age 23, male',
       'Population, age 23, total', 'Population, age 24, female',
       'Population, age 24, male', 'Population, age 24, total',
       'Population, age 25, female', 'Population, age 25, male',
       'Population, age 25, total', 'Population, age 3, female',
       'Population, age 3, male', 'Population, age 3, total',
       'Population, age 4, female', 'Population, age 4, male','Population, age 4, total', 'Population, age 5, female',
       'Population, age 5, male', 'Population, age 5, total',
       'Population, age 6, female', 'Population, age 6, male',
       'Population, age 6, total', 'Population, age 7, female',
       'Population, age 7, male', 'Population, age 7, total',
       'Population, age 8, female', 'Population, age 8, male',
       'Population, age 8, total', 'Population, age 9, female',
       'Population, age 9, male', 'Population, age 9, total',
       'Population, ages 0-14 (% of total)', 'Population, ages 0-14, female',
       'Population, ages 0-14, male', 'Population, ages 0-14, total',
       'Population, ages 10-14, female', 'Population, ages 10-14, male',
       'Population, ages 10-14, total', 'Population, ages 10-15, female',
       'Population, ages 10-15, male', 'Population, ages 10-15, total',
       'Population, ages 10-16, female', 'Population, ages 10-16, male',
       'Population, ages 10-16, total', 'Population, ages 10-17, female',
       'Population, ages 10-17, male', 'Population, ages 10-17, total',
       'Population, ages 10-18, female', 'Population, ages 10-18, male',
       'Population, ages 10-18, total', 'Population, ages 11-15, female',
       'Population, ages 11-15, male', 'Population, ages 11-15, total',
       'Population, ages 11-16, female', 'Population, ages 11-16, male',
       'Population, ages 11-16, total', 'Population, ages 11-17, female',
       'Population, ages 11-17, male', 'Population, ages 11-17, total',
       'Population, ages 11-18, female', 'Population, ages 11-18, male'],axis=1)

In [ ]:
DF_pivot.columns

ON VA MAINTENANT REGROUPER CE DATA FRAME AVEC CELUI DE Fichier_Country

In [ ]:
DF_Final=DF_pivot.merge(Fichier_Country,on='Country Code')
DF_Final

In [ ]:
Seri=DF_Final['Short Name']
DF_Final=DF_Final.drop(['Short Name'],axis=1)
DF_Final.insert(1,'Short Name',Seri)
DF_Final

In [ ]:
DF_Final_2=DF_pivot.merge(Fichier_Country[['Country Code','Short Name']],on='Country Code')
Seri=DF_Final_2['Short Name']
DF_Final_2=DF_Final_2.drop(['Short Name'],axis=1)
DF_Final_2.insert(1,'Short Name',Seri)
DF_Final_2

ON PEUT CONSTATER QUE DUE A LA SOMMATION DES INDICATERUS SUR LA PERIODE 2011 - 2015 ILS ONT PERDUS DE LEURS SENS.

ON VA MOYENNER CES VALEURS AFIN DE LEUR RENDRE LEUR PERTINENCE

In [ ]:
for col in DF_Final_2.columns:
    if col not in ['Country Code','Short Name']:
        DF_Final_2[col]=DF_Final_2[col]/5
DF_Final_2

ON VA REGROUPER LES 4 COLONNES 'ENROLMENT' EN UNE SEULE NOMMEE 'POPULATION CIBLE'

In [4036]:
DF_Final_2['Population Cible']=DF_Final_2[['Enrolment in secondary education, both sexes (number)',
                                           'Enrolment in secondary general, both sexes (number)',
                                           'Enrolment in secondary vocational, both sexes (number)',
                                           'Enrolment in tertiary education, all programmes, both sexes (number)']].sum(axis=1)


In [ ]:
DF_Final_2=DF_Final_2.drop(['Enrolment in secondary education, both sexes (number)',
                                           'Enrolment in secondary general, both sexes (number)',
                                           'Enrolment in secondary vocational, both sexes (number)',
                                           'Enrolment in tertiary education, all programmes, both sexes (number)'],axis=1)
DF_Final_2

ON VA MAINTENANT APPLIQUER UN STANDARD SCALER A NOS COLONNES NUMERIQUES AFIN DE POUVOIR PLUS AISEMENT LES COMPARER


In [ ]:
# ON COMMENCE PAR SELECTIONNER LES COLONNES A NORMALISER
colonnes_a_normaliser=[]
for col in DF_Final_2.columns:
    if col not in ['Country Code','Short Name']:
        colonnes_a_normaliser.append(col)
colonnes_a_normaliser

In [4039]:
# ON IMPORTE LE STANDARD SCALER DE LA BIBLIOTHEQUE SCIKIT LEARN
from sklearn.preprocessing import StandardScaler



In [4040]:
# ON DEFINIT NOTRE SCALER
scaler=StandardScaler()

In [4041]:
# CREATION DU NOUVEAU DF
DF_Final_2[colonnes_a_normaliser] = scaler.fit_transform(DF_Final_2[colonnes_a_normaliser])

In [ ]:
DF_Final_2

ON VA CONSERVER LES INDICATEURS SUIVANTS 'GDP, PPP (current international $)' 'Internet users (per 100 people)' 'Population Cible' 'Unemployment, total (% of total labor force)'

In [ ]:
DF_etude=DF_Final_2[['Country Code','Short Name','GDP, PPP (current international $)', 'Internet users (per 100 people)', 'Population Cible', 'Unemployment, total (% of total labor force)']]
DF_etude

ON VA AFFICHER LA DISTRIBUTION, LES BOXPLOT, 

In [ ]:
# CREATION DE LA MOSAIQUE
# LA MOSAIQUE PERMET DE FACILEMENT GERER LES POSITIONS DES SUBPLOTS
layout="""
AABB
AABB

"""

fig,axd = plt.subplot_mosaic(layout,figsize=(12,8),dpi=100)
# CREATION D'UN BOX PLOT EN UTILISANT SEABORN
sns.boxplot(y=DF_etude['GDP, PPP (current international $)'],ax=axd['A'])
axd['A'].set_title('Boxplot')
axd['A'].set_ylabel('Valeurs')

# CREATION D'UN HISTOGRAMME AVEC COURBE DE DENSITE EN UTILISANT SEABORN

sns.histplot(DF_etude['GDP, PPP (current international $)'],kde=True,ax=axd['B'])
axd['B'].set_title('Distribution')
axd['B'].set_xlabel('Valeurs')
axd['B'].set_ylabel('Frequence')

plt.tight_layout()
plt.show()